pkg/volume/csi/csi_mounter.go

```diff
@@ -377,30 +383,49 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 // 1) if fstype is "", then skip fsgroup (could be indication of non-block filesystem)
 // 2) if fstype is provided and pv.AccessMode == ReadWriteOnly and !c.spec.ReadOnly then apply fsgroup
 func (c *csiMountMgr) applyFSGroup(fsType string, fsGroup *int64, fsGroupChangePolicy *v1.PodFSGroupChangePolicy) error {
-	if fsGroup != nil {
-		if fsType == "" {
-			klog.V(4).Info(log("mounter.SetupAt WARNING: skipping fsGroup, fsType not provided"))
-			return nil
-		}
+	featureGateEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CSIVolumeSupportsFSGroup)
```

Comment:
We already checked for the feature gate outside this call. What if the featuregate check is done at the time that `c.SupportFSGroup` is set, and if the feature is disabled, set it to the default behavior "Evaluate".

Then I think the logic here could be simplified.